google/cloud/internal/grpc_request_metadata_test.cc

```diff
@@ -46,6 +47,27 @@ TEST(GrpcRequestMetadata, GetRequestMetadataFromContext) {
                                                 Pair("trailer2", "value4")));
 }
 
+TEST(GrpcRequestMetadata,
+     GetRequestMetadataFromContextWhenInitialMetadataIsNotReady) {
+  auto const server_metadata =
+      RpcMetadata{{{"header1", "value1"}, {"header2", "value2"}},
+                  {{"trailer1", "value3"}, {"trailer2", "value4"}}};
+  grpc::ClientContext context;
+  testing_util::SetServerMetadata(context, server_metadata,
+                                  /*is_initial_metadata_ready*/ false);
+
+  auto md = GetRequestMetadataFromContext(context,
+                                          /*is_initial_metadata_ready*/ false);
+  EXPECT_THAT(md.headers,
+              UnorderedElementsAre(
+                  // This function also returns the peer and compression
+                  // algorithm as synthetic headers.
+                  Pair(":grpc-context-peer", _),
+                  Pair(":grpc-context-compression-algorithm", "identity")));
+  EXPECT_THAT(md.trailers, UnorderedElementsAre(Pair("trailer1", "value3"),
+                                                Pair("trailer2", "value4")));
```

Comment:
There will be no trailers, but that is probably what happens in the real world when we don't get any headers to begin with.

Reply:
Ack.